components/rmrk/shoppingActions.ts

```diff
@@ -0,0 +1,159 @@
+import Connector from '@kodadot1/sub-api'
+import exec, { execResultValue, txCb } from '@/utils/transactionExecutor'
+import { notificationTypes, showNotification } from '@/utils/notification'
+import { unpin } from '@/utils/proxy'
+import { somePercentFromTX } from '@/utils/support'
+import nftById from '@/queries/nftById.graphql'
+
+export const enum ShoppingAction {
+  SEND = 'SEND',
+  CONSUME = 'CONSUME',
+  LIST = 'LIST',
+  BUY = 'BUY',
+}
+export const ownerActions = [
+  ShoppingAction.SEND,
+  ShoppingAction.CONSUME,
+  ShoppingAction.LIST,
+]
+export const buyActions = [ShoppingAction.BUY]
+
+export const getActions = function (
+  isOwner: boolean,
+  isAvailableToBuy: boolean
+) {
+  return isOwner ? ownerActions : isAvailableToBuy ? buyActions : []
+}
+
+const constructRmrk = (
```

Comment:
pls deprecate and use [createInteraction](https://github.com/vikiival/minimark/blob/932ddeee3e7441d0a3416f5ce8815f545dc76d37/src/rmrk/create.ts#L8)

Reply:
done